### PR TITLE
Make `Source#initialize` return a Promise

### PR DESCRIPTION
**Pull request by jmanero-r7**
This will simplify the definition of dependency chains in the manager module.

* Remove the `Metadata#clear` method. This operation souldn't be possible.
* Move more a lot of boilerplate from the Metadata module into a common base for Source classes

### CODE REVIEW

**Comment by fmitchell-r7** [comment]
👍 Thanks for refactoring this! :shipit:

File: lib/source/common.js [hunk 1]
```diff
@@ -0,0 +1,184 @@
+/* global Log, Config */
+'use strict';
+
+const EventEmitter = require('events').EventEmitter;
+const DEFAULT_INTERVAL = 60000;
+
+const deepDiff = require('deep-diff').diff;
+
+const Source = (Parser) => class extends EventEmitter {
```
Comment (dgreene-r7) [inline lib/source/common.js]: It took me a few seconds to realize what this was doing but damn that's awesome.

Comment (jmanero-r7) [inline lib/source/common.js]: Shamelessly boosted that idiom from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes#Mix-ins

**Comment by dgreene-r7** [comment]
Looks great!

**Pull request merged**